api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1857,8 +1857,98 @@ type AzurePlatformSpec struct {
+	// azure. The client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated
+	// hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	ClusterAPIAzureClientID ManagedIdentityClientID `json:"clusterAPIAzureClientID"`
+
+	// controlPlaneClientID is the client ID of a pre-existing managed identity ID associated with the control plane
+	// operator. The client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated
+	// hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	ControlPlaneClientID ManagedIdentityClientID `json:"controlPlaneClientID"`
+
+	// kmsClientID is the client ID of a pre-existing managed identity ID associated with Azure KMS. The client
+	// ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters
+	// in the form 8-4-4-4-12.
+	//
+	// +optional
+	KMSClientID ManagedIdentityClientID `json:"kmsClientID,omitempty"`
+
+	// imageRegistryClientID is the client ID of a pre-existing managed identity ID associated with the
+	// cluster-image-registry-operator. The client ID of a managed identity must be a valid UUID. It should be 5 groups
+	// of hyphen separated hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	ImageRegistryClientID ManagedIdentityClientID `json:"imageRegistryClientID"`
+
+	// ingressClientID is the client ID of a pre-existing managed identity ID associated with the
+	// cluster-ingress-operator. The client ID of a managed identity must be a valid UUID. It should be 5 groups of
+	// hyphen separated hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	IngressClientID ManagedIdentityClientID `json:"ingressClientID"`
+
+	// networkClientID is the client ID of a pre-existing managed identity ID associated with the
+	// cluster-network-operator. The client ID of a managed identity must be a valid UUID. It should be 5 groups of
+	// hyphen separated hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	NetworkClientID ManagedIdentityClientID `json:"networkClientID"`
+
+	// diskClientID is the client ID of a pre-existing managed identity ID associated with the
+	// azure-disk-controller. The client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen
+	// separated hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	DiskClientID ManagedIdentityClientID `json:"diskClientID"`
+
+	// fileClientID is the client ID of a pre-existing managed identity ID associated with the
+	// azure-disk-controller. The client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen
+	// separated hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	FileClientID ManagedIdentityClientID `json:"fileClientID"`
```

Comment:
Both os these are for the disk controller?

Reply:
Yes it runs two different controllers

Reply:
Yet you've called them both azure-disk-controller, you should call them out as different things else the descriptions would lead me to believe they are for the same thing?